src/dtos/dataset-dto.ts

```diff
@@ -142,6 +156,18 @@ export class RevisionDTO {
                 impDto.uploaded_at = imp.uploadedAt.toISOString();
                 impDto.type = imp.type;
                 impDto.location = imp.location;
+                impDto.sources = await Promise.all(
+                    (await imp.sources).map(async (source: Source) => {
+                        const sourceDto = new SourceDTO();
+                        sourceDto.id = source.id;
+                        sourceDto.import_id = (await source.import).id;
+                        sourceDto.revision_id = (await source.revision).id;
+                        sourceDto.csv_field = source.csvField;
+                        sourceDto.action = source.action;
+                        sourceDto.type = source.type;
+                        return sourceDto;
```

Comment:
Yes... although I like the control to ensure we're not nesting objects to much.  For example if you tried to return the whole dataset.  You'd ended up with multiples of the dimension objects for example as they are referenced by both revision and dataset.... Multiple sources as well because they are referenced by both import and dimension.  I do like the control of specifying exactly what we're returning from a given DTO to prevent that.

Lets have a look at this as we break down the file in to individual DTO files for each object.

Reply:
👍